packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -855,13 +895,24 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       widget.controller.addListener(_didChangeTextEditingValue);
       _updateRemoteEditingValueIfNeeded();
     }
+    if (widget.controller.selection != oldWidget.controller.selection) {
+      _selectionOverlay?.update(_value);
+    }
+    _selectionOverlay?.handlesVisible = widget.showSelectionHandles;
     if (widget.focusNode != oldWidget.focusNode) {
       oldWidget.focusNode.removeListener(_handleFocusChanged);
       _focusAttachment?.detach();
       _focusAttachment = widget.focusNode.attach(context);
       widget.focusNode.addListener(_handleFocusChanged);
       updateKeepAlive();
     }
+    if (widget.readOnly) {
+      _closeInputConnectionIfNeeded();
+    } else {
+      if (oldWidget.readOnly && _hasFocus)
+        _openInputConnection();
+    }
+    _cursorVisibilityNotifier.value = widget.showCursor;
```

Comment:
Why do we need to force this here? Wouldn't that automatically happen on the next CursorColorTick?

(not a big deal, but I would expect that we may see irregular blinking if we just randomly force this outside of the regular blink schedule)?